schema.ts

```diff
@@ -1316,6 +1321,23 @@ export interface ObjectSchemaDefinition<K extends string, L extends string>
    * @hidden
    */
   groupIdProperty?: PropertyIdentifier<K>;
+
+  /**
+   * For cases where the object being synced represents a nested group, the name of the property within
+   * {@link ObjectSchemaDefinition.properties} that identifies the id of the nested group in the service
+   * being synced from.
+   *
+   * One or both of {@link ObjectSchemaDefinition.nestedGroupIdProperty}/{@link ObjectSchemaDefinition.userIdProperty}
+   * are required for sync tables with role {@link TableRole.GroupMembers}.
+   *
+   * Must be a {@link ValueType.String} or {@link ValueType.Number} property
+   *
+   * Nested groups are additive. All children members of a nested group will be considered members of the parent group.
+   * TODO(sam): Unhide this
+   * @hidden
+   */
+
+  nestedGroupIdProperty?: PropertyIdentifier<K>;
```

Comment:
Hmm sorry I'm having trouble picturing this. Is the id here that a row in the group members table can be _either_ a user or a child group?

If so, I think that's confusing. I don't think we want tables where the rows are polymorphic. This feels better to express on a Groups table.

Reply:
The question to me is do we view parent groups as some separate thing from just group membership in general?

Reply:
Hmm I guess I was just imagining that there would be a parentIdProperty on a Groups table.

I guess I don't feel strongly. Would be good to have another pack with nested groups to pressure test this with.